clients/imodels-client-management/src/index.ts

```diff
@@ -2,6 +2,17 @@
  * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
-export * from "./base/types";
-export * from "./operations/OperationParamExports";
-export * from "./IModelsClient";
+export * from "./base/types/index.js";
+export * from "./operations/OperationParamExports.js";
+export * from "./IModelsClient.js";
+export * from "./Constants.js" // used in client-authoring
```

Comment:
i think we will need to discuss this with the imodel team & review this

Reply:
Why iTwin.js core doesn't need these changes, but they would be needed in this repo? Seems like bundler misconfiguration. There would have to be a very good reason to approve any *.js file imports/exports or barrel files (index.ts) being listed instead of directories.

Reply:
Some parts of this codebase also directly depend on cjs, so esm cannot be the main module loader and can only be generated as an output (e.g., none of the tests can work as esm and would have to be rewritten).

Reply:
which parts directly depend on cjs?

Reply:
The version of DI framework we are using hasn't been properly working with ESM before. There's newer versions of the DI, that have breaking changes and would require changes to all our repositories.